_rules/inline-link-in-paragraph-has-different-background-color-and-distinguishable-style-on-hover-and-focus-2803b8.md

```diff
@@ -0,0 +1,143 @@
+
+## Expectation
+
+For each test target all the following are true:
+
+- a [highest possible contrast][] that is at least 3.0:1 between its [background color][] and the [background color][] of the [visible][] [text nodes][text node] [rendered on the same line][rendered on a line] that are not [descendants][descendant] in the [flat tree][] of a [semantic link][]; and
+- a [distinguishing style][] both when the target element [gains focus][focused] and when the target element is [hovered][].
+
+## Assumptions
+
+- The link is distinguishable from the rest of the text with color, which means it fails SC 1.4.1 when there is not another way to distinguish it.
+- The 3:1 contrast difference between text is minimal to what would be sufficient to meet WCAG 2.0. This value is part of [technique G183](https://www.w3.org/WAI/WCAG21/Techniques/general/G183), but is not specified in the [1.4.1 success criterion](https://www.w3.org/WAI/WCAG21/Understanding/use-of-color.html).
+- Any change in font is sufficiently distinguishable, and fonts are loaded when they are present.
+- If different background colors are used by the different visible non-link text nodes then the background color can not be a distinguishing factor.
+
+## Accessibility Support
+
+_No accessibility support issues known._
+
+## Background
+
+- [Understanding Success Criterion 1.4.1: Use of Color](https://www.w3.org/WAI/WCAG21/Understanding/use-of-color.html)
+- [G183: Using a contrast ratio of 3:1 with surrounding text and providing additional visual cues on focus for links or controls where color alone is used to identify them](https://www.w3.org/WAI/WCAG21/Techniques/general/G183)
+- [F73: Failure of Success Criterion 1.4.1 due to creating links that are not visually evident without color vision](https://www.w3.org/WAI/WCAG21/Techniques/failures/F73)
+- [C15: Using CSS to change the presentation of a user interface component when it receives focus](https://www.w3.org/WAI/WCAG21/Techniques/css/C15)
+
+## Test Cases
+
+### Passed
+
+#### Passed Example 1
+
+This link, that is a descendant of a paragraph element, has text that has a background color contrast of more than 3:1 compared to the other text in the paragraph. When the link receives focus, an underline appears. When the link receives hover, an underline appears.
+
+```html
+<style>
+	a {
+		text-decoration: none;
+		color: #000;
+		background-color: #cf5e42;
+	}
+	a:hover,
+	a:focus {
+		text-decoration: underline;
+	}
+</style>
+<p>Read about WAI on the <a href="http://w3.org/WAI">WAI webpage</a>.</p>
+```
+
+### Failed
+
+#### Failed Example 1
+
+This link, that is a descendant of a paragraph element, has no visual cues of being recognized as a link with the underline removed.
+
+```html
+<style>
+	a.test {
+		text-decoration: none;
+	}
+</style>
+<p>Read about WAI on the <a class="test" href="http://w3.org/WAI">WAI webpage</a>.</p>
+```
+
```

Comment:
This failed example does not have a background colour different to the other text, so does not seem relevant for the specifics of this rule.

There should be a failed example where the link text has a background colour different from the other text but with insufficient contrast. And a failed example without focus and/or hover effect. Possibly as 3 separate failed examples.

Reply:
Good suggestions.